packages/modules/pricing/src/services/pricing-module.ts

```diff
@@ -1402,10 +1416,12 @@ export default class PricingModuleService
 
     const priceListsUpsert = priceLists
       .map((priceList) => {
+        const priceListRules =
+          priceList.price_list_rules as unknown as Collection<
+            InferEntityType<typeof PriceListRule>
+          >
```

Comment:
@adrien2p Here is the place where we have to re-convert the type to MikroORM collection. The same thing we talked about during our initial discussions

Reply:
we shouldnt have to do that in the service module layer, we can manipulate it as a classic array and loop through it as it implements the iterator symbol as well. the toArray will apply toJSON on it but I don't think we care much here for that